bottlecap/src/lifecycle/invocation/triggers/step_function_event.rs

```diff
@@ -0,0 +1,366 @@
+        let json = read_json_file("step_function_legacy_event.json");
+        let payload = serde_json::from_str(&json).expect("Failed to deserialize StepFunctionEvent");
+
+        assert!(StepFunctionEvent::is_match(&payload));
+    }
+
+    #[test]
+    fn test_is_not_match() {
+        let json = read_json_file("sqs_event.json");
+        let payload = serde_json::from_str(&json).expect("Failed to deserialize SqsRecord");
+        assert!(!StepFunctionEvent::is_match(&payload));
+    }
+
+    #[test]
+    fn test_get_tags() {
+        let json = read_json_file("step_function_event.json");
+        let payload = serde_json::from_str(&json).expect("Failed to deserialize into Value");
+        let event =
+            StepFunctionEvent::new(payload).expect("Failed to deserialize StepFunctionEvent");
+        let tags = event.get_tags();
+
+        let expected = HashMap::from([(
+            "function_trigger.event_source".to_string(),
+            "states".to_string(),
+        )]);
+
+        assert_eq!(tags, expected);
+    }
+
+    #[test]
+    fn test_get_arn() {
+        let json = read_json_file("step_function_event.json");
+        let payload = serde_json::from_str(&json).expect("Failed to deserialize into Value");
+        let event =
+            StepFunctionEvent::new(payload).expect("Failed to deserialize StepFunctionEvent");
+        assert_eq!(
+            event.get_arn("us-east-1"),
+            "arn:aws:states:us-east-1:425362996713:stateMachine:agocsTestSF"
+        );
+    }
+
+    #[test]
+    fn test_get_carrier() {
+        let json = read_json_file("step_function_event.json");
+        let payload = serde_json::from_str(&json).expect("Failed to deserialize into Value");
+        let event =
+            StepFunctionEvent::new(payload).expect("Failed to deserialize StepFunctionEvent");
+        let carrier = event.get_carrier();
+
+        let expected = HashMap::new();
+
+        assert_eq!(carrier, expected);
+    }
+
+    #[test]
+    fn get_span_context() {
+        let json = read_json_file("step_function_event.json");
+        let payload = serde_json::from_str(&json).expect("Failed to deserialize into Value");
+        let event =
+            StepFunctionEvent::new(payload).expect("Failed to deserialize StepFunctionEvent");
+
+        let span_context = event.get_span_context();
+
+        let expected = SpanContext {
+            trace_id: 5744042798732701615,
+            span_id: 3347375825762253425,
+            sampling: Some(Sampling {
+                priority: Some(1),
+                mechanism: None,
+            }),
+            origin: Some("states".to_string()),
+            tags: HashMap::from([(
+                DATADOG_HIGHER_ORDER_TRACE_ID_BITS_KEY.to_string(),
+                "1914fe7789eb32be".to_string(),
+            )]),
+            links: vec![],
+        };
+
+        assert_eq!(span_context, expected);
+    }
+
+    #[test]
+    fn test_generate_parent_id() {
```

Comment:
These tests are the ones from the Go Agent, so we can see that data is generated as expected.